exportoptionsgenerator/exportoptionsgenerator.go

```diff
@@ -91,6 +92,10 @@ func (g ExportOptionsGenerator) GenerateApplicationExportOptions(
 		exportOpts = addManualSigningFields(exportOpts, codeSignGroup, archivedWithXcodeManagedProfiles, g.logger)
 	}
 
+	if testFlightInternalTestingOnly {
```

Comment:
It seems this option was introduced in Xcode 15 (https://github.com/fastlane/fastlane/discussions/21525), I suggest setting this option only if xcodeMajorVersion >= 15 similarly to `addDistributionBundleIdentifierFromXcode12`.